commons/zenoh-protocol-core/src/whatami.rs

```diff
@@ -9,14 +11,14 @@ pub enum WhatAmI {
 }
 
 impl std::str::FromStr for WhatAmI {
-    type Err = ();
+    type Err = ZError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         match s {
             "router" => Ok(WhatAmI::Router),
             "peer" => Ok(WhatAmI::Peer),
             "client" => Ok(WhatAmI::Client),
-            _ => Err(()),
+            _ => bail!("{} is not one of the router, peer, or client.", s),
```

Comment:
I suggest to change the error message as follows:

bail!("{} is not a valid WhatAmI value. Valid values are: [\"router\", \"peer\", \"client\"]., s)

Reply:
Okay, let me change the writing. Thanks!